### PR TITLE
Add open graph and ld+json parsers

### PR DESCRIPTION
**Pull request by ukalbhor**
DTSHOPSDK-559

5386d0e Add open graph tag parser
46dde52 Add ldjson parser
f7d1c03 Add tryAndLog helper method

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -5,6 +5,8 @@
 .*/dist/module
 ./build-identity-iframe.js
 ./iframes/identity/identity.js
+.*/src/lib/tag-parsers/.*
```
Comment (batiskafff) [inline .flowconfig:8]: Why did you decided to ignore this folder?

File: src/lib/utils.js [hunk 2]
```diff
@@ -46,4 +46,12 @@ const autoGenerateProductPayload = () : ?GeneratedProductPayload => {
   return Object.keys(attributes).length > 0 ? attributes : null;
 };
 
-export default autoGenerateProductPayload;
+export const tryAndLog = (fn : Function) => {
+  return () => {
+    try {
+      return fn();
+    } catch (err) {
+      console.log(err); /* eslint-disable-line no-console */
```
Comment (batiskafff) [inline src/lib/utils.js:54]: Do we need it browser console?

File: test/lib/tag-parsers/res/fanatics_product.html [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
...l get you pumped up and ready for game day."/><meta property="twitter:title" content="LeBron James Los Angeles Lakers Jordan Brand 2021/22 #6 Swingman Player Jersey Purple - Statement Edition"/><meta property="twitter:description" content="This Jordan Brand LeBron James Swingman Player Jersey will make any loyal Los Angeles Lakers fan feel like a part of the team. It features crisp Los Angeles Lakers graphics and Dri-FIT technology to provide a bold yet comfortable look. This LeBron James gear will get you pumped up and ready for game day."/><meta property="og:type" content="product"/><meta property="twitter:card" content="summary_large_image"/><meta property="twitter:site" content="@Fanatics"/><meta property="og:site_name" content="Fanatics"/><meta property="og:url" content="https://www.fanatics.com/nba/los-angeles-lakers/lebron-james-los-angeles-lakers-jordan-brand-2021/22-number-6-swingman-player-jersey-purple-statement-edition/o-3536+t-58252968+p-60467601075+z-9-1901122059"/><meta property="og:price:amount" content="109.99"/><meta property="og:price:currency" content="USD"/><meta property="og:availability" content="In stock"/><meta name="theme-color" content="#FFFFFF"/><script>
```
Comment (batiskafff) [inline test/lib/tag-parsers/res/fanatics_product.html:2]: It looks little bit overkill, might be we can have smaller version of this web page?

**Review by batiskafff — APPROVED** [review]

File: .flowconfig (hunk 1 above)
Comment (ukalbhor) [reply .flowconfig:8]: The parsers deal with looking up DOM HTml elements and working with HTMLTypes in flow is a PITA

File: src/lib/utils.js (hunk 2 above)
Comment (ukalbhor) [reply src/lib/utils.js:54]: @dturgumbaev what do you suggest ?

**Comment by ukalbhor** [comment]
Merging. This is independent code currently not invoked from the SDK. 
We can re-review in the next PR that has invocation of the parsers.

**Pull request merged**